system/Database/MySQLi/Connection.php

```diff
@@ -368,11 +368,17 @@ public function escapeLikeStringDirect($str)
     /**
      * Generates the SQL for listing tables in a platform-dependent manner.
      * Uses escapeLikeStringDirect().
+     *
+     * @param false|string $tableName
      */
-    protected function _listTables(bool $prefixLimit = false): string
+    protected function _listTables(bool $prefixLimit = false, $tableName = false): string
     {
         $sql = 'SHOW TABLES FROM ' . $this->escapeIdentifiers($this->database);
 
+        if ($tableName !== false) {
+            return $sql . " LIKE " . $this->escape($tableName);
```

Comment:
If you use LIKE, escape the value with `escapeLikeString()`.

Reply:
I'm not using LIKE in the traditional manner with a wildcard.

Reply:
Not sure what the difference is.

Reply:
Well not sure what to do about it.  Do I need to create a mock class?

Honestly, I think scrapping the mock test and doing a live test instead would be better.